### PR TITLE
add CarControllerBase base class to all CarControllers

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

File: selfdrive/car/interfaces.py [hunk 1]
```diff
@@ -518,3 +528,12 @@ def predict(self, x: list[float], do_sample: bool = False):
...
     return pred
+
+
+SendCan = tuple[int, int, bytes, int]
+
+
+class CarControllerBase(CarInterfaceHelpers, ABC):
+  @abstractmethod
+  def update(self, CC, CS, now_nanos) -> tuple[car.CarControl.Actuators, list[SendCan]]:
+    pass
```
Comment (sshane) [inline selfdrive/car/interfaces.py:539]: Should be another PR, we can clean up more duplicate code with a base CC class, we just never got around to it

**Comment by sshane** [comment]
Pattern of subclassing to add functionality to the car interfaces seems odd, don't know if we need this.

File: selfdrive/car/interfaces.py (hunk 1 above)
Comment (jnewb1) [reply selfdrive/car/interfaces.py]: ill just add the base class for now
